src/Hie/Implicit/Cradle.hs

```diff
@@ -67,7 +52,7 @@ implicitConfig' fp =
     --   <|> (Bazel,) <$> rulesHaskellWorkDir fp
     <|> (cabalExecutable >> cabalProjectDir fp >> cabalDistDir fp >>= cabal)
     <|> (stackExecutable >> stackYamlDir fp >> stackWorkDir fp >>= stack)
-    <|> (cabalExecutable >> cabalProjectDir fp >>= cabal)
+    <|> (cabalExecutable >> (cabalProjectDir fp <|> cabalDistDir fp) >>= cabal)
```

Comment:
This will change the behavior of ghcide and hls.
The fixed behavior matches `gen-hie`.

The old behavior uses a stack cradle in the implicit-hie repo, unless I delete it's `stack.yaml`.